### PR TITLE
Improve logging of dangerous userdata edge cases

### PR DESCRIPTION
**Pull request by mdcfe**
### Information

This PR does not close an existing issue. (todo: link relevant issues)

### Details

**Proposed feature:**  
This commit improves the logging of a few edge cases that can lead to loss of userdata:
- Third-party plugins forcing creation of a NPC account which could not be found on the usermap, even if the account file exists
- UserMap#trackUUID being called with a conflicting UUID but replace set to false

**Environments tested:**    

OS: Windows 10 20H2 (19042)

Java version: 11.0.7+10

- [x] Most recent Paper version (1.16.5, git-Paper-466)
- [ ] ~~CraftBukkit/Spigot/Paper 1.12.2~~
- [ ] ~~CraftBukkit 1.8.8~~

### CODE REVIEW

File: Essentials/src/main/java/com/earth2me/essentials/UserMap.java [hunk 1]
```diff
@@ -132,9 +132,7 @@ public void trackUUID(final UUID uuid, final String name, final boolean replace)
                         names.put(keyName, uuid);
                         uuidMap.writeUUIDMap();
                     } else {
-                        if (ess.getSettings().isDebug()) {
-                            ess.getLogger().info("Found old UUID for " + name + " (" + uuid.toString() + "). Not adding to usermap.");
-                        }
+                        ess.getLogger().info("Found UUID " + uuid.toString() + " for player " + name + ", but player already has a UUID (" + names.get(keyName).toString() + "). Not replacing UUID in usermap.");
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/UserMap.java]: Exception

File: Essentials/src/main/java/com/earth2me/essentials/api/Economy.java [hunk 2]
```diff
@@ -879,6 +890,7 @@ public static boolean createNPC(final String name) {
             createNPCFile(name);
             return true;
         }
+        LOGGER.warning(MessageFormat.format(WARN_EXISTING_NPC_CREATE, name, user.getConfigUUID()));
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/api/Economy.java]: Exception

File: Essentials/src/main/java/com/earth2me/essentials/api/Economy.java [hunk 3]
```diff
@@ -79,7 +90,7 @@ private static User getUserByName(final String name) {
             if (player != null) {
                 user = ess.getUser(player.getUniqueId());
                 if (user != null) {
-                    logger.info(String.format("[Economy] Found player %s by UUID %s but not by their actual name - they may have changed their username", name, player.getUniqueId().toString()));
+                    LOGGER.info(MessageFormat.format(WARN_PLAYER_UUID_NO_NAME, name, player.getUniqueId().toString()));
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/api/Economy.java]: Exception

**Review by JRoy — CHANGES_REQUESTED** [review]
Exception stack traces are going to be useful in the process of debugging what plugins are causing this

**Review by JRoy — APPROVED** [review]

**Comment by mibby** [comment]
@JRoy Essentials blew up with exceptions when someone typed ``/baltop``. Seems to happen every time you type it as well. Using EssentialsX dev 1112 (2.19.0-dev+70-34fdcf8).

https://paste.gg/p/anonymous/c91d995181bb4b9b87b2f47009794dc0/files/40922ecedf7c43f2bd32cbb9f4439e8a/raw

**Comment by JRoy** [comment]
@mibby can you open a bug report with that so we can properly track it. (Make sure you fill out the whole template) It is intended so we can find bugs with our UserMap

**Pull request merged**